app/models/vulnerability.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: vulnerabilities
+#
+#  id         :integer          not null, primary key
+#  name       :string(255)      not null
+#  scanner    :string(255)      default(""), not null
+#  severity   :string(255)      default(""), not null
+#  link       :string(255)      default(""), not null
+#  fixed_by   :string(255)      default(""), not null
+#  metadata   :text(65535)
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+# Indexes
+#
+#  index_vulnerabilities_on_name  (name) UNIQUE
+#
+
+# Vulnerability represents a vulnerability as detected by any of our sources.
+class Vulnerability < ActiveRecord::Base
+  has_many :scan_results, dependent: :destroy
+  has_many :tags, -> { uniq }, through: :scan_results
+
+  validates :name, uniqueness: true
+
+  # Add some fields into this vulnerability object given an `obj` with the
+  # `sc` scanner response.
+  def add_extra_values!(obj:, sc:)
+    attrs = fetch_attributes(obj: obj, sc: sc)
+    update(attrs) unless attrs.empty?
+  end
+
+  # With the given objects, fetch the attributes to be updated for this object.
+  def fetch_attributes(obj:, sc:)
+    attrs = {}
+    attrs[:fixed_by] = obj["FixedBy"] if obj["FixedBy"].present?
+    attrs[:link]     = obj["Link"] if obj["Link"].present?
+    attrs[:metadata] = obj["Metadata"] if obj["Metadata"].present?
+    attrs[:scanner]  = sc if scanner.blank?
+    attrs[:severity] = obj["Severity"] if obj["Severity"].present?
+    attrs
```

Comment:
Maybe also add description? I guess it might be as useful as metadata if we decide to enhance the vulnerabilities page. Or the description field is not the one explaining what the vulnerability is about?

Reply:
I am not sure. Imaging the next step, if we want to process the results. e.g whitelist, ignore vulnerabilities or interact with them in any other way. I would make sense to add a description to that particular table. If that is not planned then description is a good fit. From the architecture perspective it makes more sense to have single purpose tables. Its easier to extend and add other features.

Reply:
@Vad1mo so @vitoravelino proposed that we add the description in the new vulnerabilities table as a  new column. I agree with that, but I don't quite understand what you mean :confused: